sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -79,7 +79,7 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
         client.connection_policy, global_endpoint_manager, *args
     )
     # HttpRequest we would need to modify for Container Recreate Retry Policy
-    request: Optional[HttpRequest] = None
+    request = None
```

Comment:
I guess the error is gone now, but do you remember what the issue was with this? I would have thought keeping it more specific would have been better

Reply:
So this one used to complain because we would call attributes from HttpRequest without explicitly saying it was going to be httprequest type, but now the issue is that by default in a function that is untyped that the body of it doesn't check the typing, So in order for us to start doing typing we need to to typehint the function itself. I am not sure if it is currently left untyped by design, but I imagine typing it would require some minor changes to the method in case of any mypy issues that may pop up.

Reply:
hmmm I see - that sucks but oh well, can you please create a follow-up to this PR for us to do better typehinting in the SDK? It's something that we should be moving towards as opposed to away from to ensure we're following best practices/ being as clear as possible with the code